File: 5_kyu/RGB_To_Hex_Conversion.go
package kata

import "fmt"

// Trim integer x to intervall min..max
func trim(x, min, max int) int {
	if x < min {
		return min
	}
	if x > max {
		return max
	}
	return x
}

func RGB(r, g, b int) string {
	r = trim(r, 0, 0xff)
	g = trim(g, 0, 0xff)
	b = trim(b, 0, 0xff)
	return fmt.Sprintf("%02X%02X%02X", r, g, b)
}

File: 5_kyu/RGB_To_Hex_Conversion.r
# Converts RGB values to hexstring  
rgb <- function(r, g, b) sprintf("%02X%02X%02X",clamp(r),clamp(g),clamp(b))

# Clamp value x t orange 0..255.
clamp <- function(x) max(0,min(x,255))
